cloudbridge/cloud/providers/gce/resources.py

```diff
@@ -1311,11 +1311,26 @@ def name(self, value):
 
     @property
     def description(self):
-        return self._volume.get('description')
+        self.refresh()
+        labels = self._volume.get('labels')
+        if not labels or 'description' not in labels:
+            return ''
+        return labels.get('description', '')
 
     @description.setter
     def description(self, value):
-        raise NotImplementedError('Not supported by this provider.')
+        self.refresh()
```

Comment:
Same here as above. This actually has me wondering why is the refresh here to begin with since the value will be set (hence, overwritten) in the next couple of lines?